[CI] Move Travis CI job "syncing available extensions to Microsoft/azure-docs-cli" to ADO

haroldrandom opened the pull request:
This is the last move for CI that migrate from Travis to ADO #1140

This PR is about updating extension list for Microsoft/azure-docs-clis

Basically, the origin job in Travis is do update a file `docs-ref-conceptual/azure-cli-extensions-list.md` of repo Microsoft/azure-docs-cli for syncing all of available extensions.
https://github.com/Azure/azure-cli-extensions/blob/f380e3ccddd9d26c84ea8ca414052aca90dc56ec/.travis.yml#L38-L51

The original scripts to do this job is placed in `scripts/avail-ext-doc/`. The most important script `main.py` do a list of jobs: 
1. clone Microsoft/azure-docs-cli
2. check whether need to update or not
3. update `docs-ref-conceptual/azure-cli-extensions-list.md`
4. save changes to remote repo
5. create PR.

Which also involves a token with write permission of Microsoft/azure-docs-cli. (we don't have it)

Here, I only pick up the 3rd step and left the other parts of Job in an **internal** pipeline "**Azure CLI Extension Sync**" in AOD. The other files are basically copied from the original.

Those PR:
- https://github.com/MicrosoftDocs/azure-docs-cli/pull/1823
- https://github.com/MicrosoftDocs/azure-docs-cli/pull/1824

Why turns it into a Pipeline?
Scheduled configuration of ADO doesn't support to apply on task level, but pipeline level. Meanwhile, the original job in Travis is also in publish stage, which is the pipeline  counterpart of ADO .

are the output of this Pipeline and get valididated.
I will migrate Github account from myself to azclibot soon.

---

This checklist is used to make sure that common guidelines for a pull request are followed.

### General Guidelines

- [x] Have you run `azdev style <YOUR_EXT>` locally? (`pip install azdev` required)
- [x] Have you run `python scripts/ci/test_index.py -q` locally?

For new extensions:

- [ ] My extension description/summary conforms to the [Extension Summary Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/extensions/extension_summary_guidelines.md).

azuresdkci commented:
If this PR is for a new extension or change to an existing extension, use the following to try out the changes in this PR:
```
docker run -it microsoft/azure-cli:latest
export EXT=<NAME>
pip install --upgrade --target ~/.azure/cliextensions/$EXT "git+https://github.com/haroldrandom/azure-cli-extensions.git@move-ext-list-to-ado#subdirectory=src/$EXT&egg=$EXT"
```

haroldrandom commented:
This PR (https://github.com/MicrosoftDocs/azure-docs-cli/pull/1825) is created by azclibot, I have already changed the Github account to azclibot.

Also, I modified the author and manager to our guys. @yungezz  @yonzhan 
![image](https://user-images.githubusercontent.com/14357159/75134928-c8ed9100-571a-11ea-839c-a0cde9fbb0da.png)

yungezz commented on scripts/ci/avail-ext-doc/list-template.md:
```diff
@@ -0,0 +1,26 @@
+description: A complete list of the officially supported extensions for the Azure CLI.
+author: haroldrandom
+ms.author: jianzen
+manager: yonzhan,yungez
```
github id: yungezz. thanks.

haroldrandom replied:
Updated.

haroldrandom commented:
This PR(https://github.com/MicrosoftDocs/azure-docs-cli/pull/1826) is triggered by schedule cron.
Build ID is 269103 

![image](https://user-images.githubusercontent.com/14357159/75139066-b0837380-5726-11ea-9411-97e12ebc1545.png)

haroldrandom commented:
After the triggered PR (https://github.com/MicrosoftDocs/azure-docs-cli/pull/1826) is merged, the pipeline won't procedure anymore when it detects no new extensions need to be updated.
![image](https://user-images.githubusercontent.com/14357159/75144924-ff370a80-5732-11ea-85e7-7130d83382d2.png)

fengzhou-msft commented:
So you plan to drop the usage of docker, just use vm agent on ADO to do the task?

haroldrandom commented:
> So you plan to drop the usage of docker, just use vm agent on ADO to do the task?

Yeah, using docker is unnecessary. The Dockerfile is dummy.
